### ISSUE TITLE
[FEA]  [SPARK-35780][SQL] Support DATE/TIMESTAMP literals across the full range 

### ISSUE THREAD

**Issue by andygrove** [NVIDIA/spark-rapids#3406]
**Is your feature request related to a problem? Please describe.**
There are quite a few changes in https://github.com/apache/spark/commit/c9813f74e98ea42d86cdb86c0eb6f989809a4237 that we need to review/

**Describe the solution you'd like**
We should support the same ranges as Spark 3.2

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by sameerz** [comment]
@revans2 is this resolved by PR #3531 ?

**Comment by sperlingxx** [comment]
> @revans2 is this resolved by PR #3531 ?

Yes,  it is related to #3382 which has been "fixed" in temporary. To address this issue, we need a lot modifications on native layer, which can be hardly done in a short time.

**Comment by revans2** [comment]
We can close this and file a follow on issue to implement it fully, or we can just keep this open and re-prioritize it based off of wanting to support this properly.

**Comment by sameerz** [comment]
Let's leave this open for fixing at a later date.  I will add it back the triage label so we discuss it before we decide how to disposition it.
